crates/cast/src/cmd/run.rs

```diff
@@ -287,6 +295,38 @@ impl RunArgs {
     }
 }
 
+pub async fn fetch_contracts_bytecode_from_trace(
+    provider: &RootProvider<AnyNetwork>,
+    result: &TraceResult,
+) -> Result<HashMap<Address, Bytes, RandomState>> {
+    let mut contracts_bytecode = HashMap::new();
+    if let Some(ref traces) = result.traces {
+        let addresses = gather_trace_addresses(traces);
+        for address in addresses {
+            let code = provider.get_code_at(address).await?;
+            if !code.is_empty() {
+                contracts_bytecode.insert(address, code);
+            }
+        }
+    }
+    Ok(contracts_bytecode)
+}
+
+fn gather_trace_addresses(traces: &Traces) -> Vec<Address> {
+    let mut addresses = HashSet::new();
+    for (_, trace) in traces {
+        for node in trace.arena.nodes() {
+            if !node.trace.address.is_zero() {
+                addresses.insert(node.trace.address);
+            }
+            if !node.trace.caller.is_zero() {
+                addresses.insert(node.trace.caller);
+            }
+        }
+    }
+    addresses.into_iter().collect()
+}
```

Comment:
@zerosnacks imo we could even upstream this as a helper to revm-inspectors

but for now this seems fine here